### PR TITLE
Basic log scope support in ConsoleExporter

### PR DESCRIPTION
**Pull request by reyang**
## Changes

Added log scope support to the ConsoleExporter

### CODE REVIEW

File: docs/logs/extending-the-sdk/MyExporter.cs [hunk 1]
```diff
@@ -42,7 +42,9 @@ public override ExportResult Export(in Batch<LogRecord> batch)
                 sb.Append(", ");
             }
 
-            sb.Append($"{record}");
+            sb.Append($"{record}(");
+            record.ForEachScope((scope, state) => sb.Append($"{scope}"), record.State);
```
Comment (CodeBlanch) [inline docs/logs/extending-the-sdk/MyExporter.cs:46]: Does this compile?

```csharp
record.ForEachScope((scope, _) => sb.Append($"{scope}"), null);
```

The `state` param is there so (more or less) you don't need a closure on the delegate. The use case for state is more like...

```csharp
List<object> scopes = new List<object>();
record.ForEachScope((scope, state) => state.Add(scope), scopes);
```

It's fine as-is, just a bit misleading, so I was trying to find a way to make it more obvious that our example isn't using the state param.

Comment (reyang) [reply docs/logs/extending-the-sdk/MyExporter.cs]: We need this in order to make it compile:

```csharp
record.ForEachScope((scope, _) => sb.Append($"{scope}"), (object)null);
```

Not C# expert, please advise if there a better way than `(object)null`.

Comment (CodeBlanch) [reply docs/logs/extending-the-sdk/MyExporter.cs]: Oh use the StringBuilder as the state! Sorry I missed that before.

```csharp
record.ForEachScope((scope, builder) => builder.Append($"{scope}"), sb);
```

Nice demo now of how it is intended to work.

You could also show static local func like:

```csharp
{
   record.ForEachScope(ProcessScope, sb);

   static void ProcessScope(object scope, StringBuilder builder)
      => builder.Append($"{scope}")
}
```

Comment (reyang) [reply docs/logs/extending-the-sdk/MyExporter.cs]: The static local func looks awesome!

File: src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs [hunk 2]
```diff
@@ -62,6 +62,13 @@ public override ExportResult Export(in Batch<LogRecord> batch)
                     this.WriteLine($"{"LogRecord.Exception:".PadRight(rightPaddingLength)}{logRecord.Exception?.Message}");
                 }
 
+                logRecord.ForEachScope(
+                    (scope, state) =>
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs:66]: TODO for myself: avoid creating a closure here.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**